public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -473,15 +491,19 @@ export class GraphiteDatasource extends DataSourceApi<GraphiteQuery, GraphiteOpt
       httpOptions.params.until = this.translateTime(options.range.to, true, options.timezone);
     }
 
-    return this.doGraphiteRequest(httpOptions).then((results: any) => {
-      if (results.data) {
-        return _.map(results.data, (tag) => {
-          return { text: tag };
-        });
-      } else {
-        return [];
-      }
-    });
+    return this.doGraphiteRequest(httpOptions)
+      .pipe(
+        map((results: any) => {
+          if (results.data) {
+            return _.map(results.data, (tag) => {
+              return { text: tag };
+            });
+          } else {
+            return [];
+          }
+        })
```

Comment:
This code is shared between `getTagsAutoComplete` and `getTagValuesAutoComplete`, maybe we can create a `OperatorFunction` both can share?